edx_exams/apps/core/api.py

```diff
@@ -158,6 +158,25 @@ def get_exam_attempt_time_remaining(exam_attempt, now=None):
     return secs_to_end
 
 
+def check_if_exam_timed_out(exam_attempt):
+    """
+    If an exam attempt's time remaining = 0 AND is in-progress/not submitted,
+    change the status of an exam attempt to 'submitted', then return attempt_id.
+    Return None otherwise.
+    """
+    IN_PROGRESS_STATUSES = [
+        ExamAttemptStatus.started,
+        ExamAttemptStatus.ready_to_submit,
+    ]
+    # NOTE: This first conditional is a dirty workaround to `get_exam_attempt_time_remaining`
+    # returning 0 when start_time or time_limit_mins is None.
+    if exam_attempt.start_time is not None and exam_attempt.allowed_time_limit_mins is not None:
```

Comment:
I'm not sure this check is needed. Any attempt that is `started` should have a start time and a set allowed time limit. If you're running into issues within `get_exam_attempt_time_remaining` because these fields are not set, I would suggest changing the order of the `if` statement, so it will short circuit. So you could do something like 

`if exam_attempt.status in IN_PROGRESS_STATUSES and get_exam_attempt_time_remaining(exam_attempt) == 0:`